src/main/java/org/folio/inventory/storage/external/ExternalStorageModuleItemCollection.java

```diff
@@ -130,8 +130,8 @@ protected JsonObject mapToRequest(Item item) {
       ? item.id
       : UUID.randomUUID().toString());
 
-    if(item.getStatus().getString(Status.NAME_KEY) != null) {
-      itemToSend.put("status", item.getStatus());
+    if(item.getStatus().getName() != null) {
```

Comment:
Should this stop the status being written out?

What happens if the item has no status when it reaches storage?

Reply:
It is a redundant check I think, API should require status to be set.

Reply:
Can there be a check in the item that enforces that?

Reply:
(This is why I like languages with non-nullable types)

Reply:
Do you mean to add a check to Item constructor that will throw an exception in case no status provided?

Reply:
Yes, effectively, I mean stop construction of items without a status. 

(If we had a language that supported it, I'd make it a closed or non-nullable type)

Reply:
I have added `requireNonNull` to the constructor of Item and Status. It is not possible to create both item or status without status and status.name now.

Is it fine?

Reply:
I think that's fine